tests/suites/cloud/suite.js

```diff
@@ -0,0 +1,237 @@
+      },
+    });
+
+    this.suite.context.set({
+      os: new BalenaOS(
+        {
+          deviceType: this.suite.deviceType.slug,
+          network: this.suite.options.balenaOS.network,
+        },
+        this.getLogger()
+      ),
+    });
+
+    // unpack OS
+    await this.context.get().os.fetch({
+      type: this.suite.options.balenaOS.download.type,
+      version: this.suite.options.balenaOS.download.version,
+    });
+
+    // get config.json for application
+    let config = await this.context
+      .get()
+      .cloud.balena.models.os.getConfig(this.context.get().balena.application, {
+        version: this.context.get().os.contract.version,
+      });
+
+    config.uuid = this.context.get().balena.uuid;
+
+    //register the device with the application, add the api key to the config.json
+    let deviceApiKey = await this.context
+      .get()
+      .cloud.balena.models.device.register(
+        this.context.get().balena.application,
+        this.context.get().balena.uuid
+      );
+    config.deviceApiKey = deviceApiKey.api_key;
+
+    // get newly registered device id, add to config.json
+    await Bluebird.delay(1000 * 10);
+    let devId = await this.context
+      .get()
+      .cloud.balena.models.device.get(this.context.get().balena.uuid);
+    config.deviceId = devId.id;
+
+    // get ready to populate DUT image config.json with the attributes we just generated
+    this.context.get().os.addCloudConfig(config);
+
+
+    // Teardown the worker when the tests end
+    this.suite.teardown.register(() => {
+      this.log("Removing image");
+      fse.unlinkSync("/data/image"); // Delete the unpacked an modified image from the testbot cache to prevent use in the next suite
+      this.log("Worker teardown");
+      return this.context.get().worker.teardown();
+    });
+
+
+    // preload image with the single container application
+    this.log(`Device uuid should be ${this.context.get().balena.uuid}`)
+    this.log("Preloading image...");
+    console.log(`initial commit is: ${initialCommit}`)
+    await this.context.get().os.configure();
+    await this.context.get().cli.preload(this.context.get().os.image.path, {
+      app: this.context.get().balena.application,
+      commit: initialCommit,
+      pin: true,
+    });
+
+    this.log("Setting up worker");
+    await this.context
+      .get()
+      .worker.network(this.suite.options.balenaOS.network);
+    
+    await this.context.get().worker.off();
+    await this.context.get().worker.flash(this.context.get().os.image.path);
+    await this.context.get().worker.on();
+
+    this.log("Waiting for device to be reachable");
+    await this.context.get().utils.waitUntil(async() => {
+      this.log("Seeing if on dashboard");
+      return await this.context
+        .get()
+        .cloud.balena.models.device.isOnline(this.context.get().balena.uuid) === true;
+    });
+
+    this.log("Device is on dashboard");
```

Comment:
```suggestion
    this.log("Device is online and provisioned successfully.");
```
Since the test can't see the dashboard visually, mentioning the UI component might be misleading.